operator/builtin/input/tcp/tcp.go

```diff
@@ -49,17 +50,21 @@ func init() {
 func NewTCPInputConfig(operatorID string) *TCPInputConfig {
 	return &TCPInputConfig{
 		InputConfig: helper.NewInputConfig(operatorID, "tcp_input"),
+		Multiline:   helper.NewMultilineConfig(),
+		Encoding:    helper.NewEncodingConfig(),
 	}
 }
 
 // TCPInputConfig is the configuration of a tcp input operator.
 type TCPInputConfig struct {
 	helper.InputConfig `yaml:",inline"`
 
-	MaxBufferSize helper.ByteSize         `json:"max_buffer_size,omitempty" yaml:"max_buffer_size,omitempty"`
-	ListenAddress string                  `json:"listen_address,omitempty" yaml:"listen_address,omitempty"`
-	TLS           *helper.TLSServerConfig `json:"tls,omitempty" yaml:"tls,omitempty"`
-	AddAttributes bool                    `json:"add_attributes,omitempty" yaml:"add_attributes,omitempty"`
+	MaxBufferSize helper.ByteSize         `mapstructure:"max_log_size,omitempty"          json:"max_log_size,omitempty"         yaml:"max_log_size,omitempty"`
```

Comment:
nit: For the sake of readability, should we change the field name to `MaxLogSize`?